src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsResourceIntTest.kt

```diff
@@ -1219,6 +1241,33 @@ class AdjudicationsResourceIntTest : IntegrationTestBase() {
           }
         }
       }
+
+      @Test
+      fun `can create adjudication with witnesses`() {
+        webTestClient.post().uri("/prisoners/$offenderNo/adjudications")
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ADJUDICATIONS")))
+          .contentType(MediaType.APPLICATION_JSON)
+          .body(
+            BodyInserters.fromValue(
+              aSimpleAdjudication(
+                adjudicationNumber = adjudicationNumber.toString(),
+                staffWitnessesUsernames = listOf(staffWitnessAdaeze.username, staffWitnessAbiodun.username),
+              ),
+            ),
+          )
+          .exchange()
+          .expectStatus().isOk
+
+        repository.runInTransaction {
+          incident = repository.getAdjudicationIncidentByAdjudicationNumber(adjudicationNumber)
+
+          assertThat(incident).isNotNull
+          val witnesses = incident!!.parties.filter { it.incidentRole == witnessRole }
+          assertThat(witnesses).hasSize(2)
+          assertThat(witnesses).anyMatch { it.staff?.id == staffWitnessAdaeze.staff.id }
+          assertThat(witnesses).anyMatch { it.staff?.id == staffWitnessAbiodun.staff.id }
```

Comment:
You could do this in one line with some `extracting` and `containsExactlyInAnyOrder`... but I suspect you chose not to 😄